MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -383,9 +383,9 @@ static inline CIE1931XY VkXYColorEXTToCIE1931XY(VkXYColorEXT xy) {
 }
 
 VkResult MVKSwapchain::getRefreshCycleDuration(VkRefreshCycleDurationGOOGLE *pRefreshCycleDuration) {
-#if MVK_IOS
+#if MVK_IOS_OR_TVOS
 	NSInteger framesPerSecond = 60;
-	if (@available(iOS 10.3, *)) {
+	if (@available(iOS 10.3, tvOS 10.3, *)) {
```

Comment:
Upgrade to latest MoltenVK master. It turns out `@available()` is fragile, and we no longer use it. See the replacements.

Reply:
I see, so you're moving to `respondsToSelector` checks instead? I had trouble with some of the other instances of that pattern in this diff. Xcode complained until I `ifdef`d out the references to `renderTargetArrayLength` and `inputPrimitiveTopology`

I will take another look. Perhaps it's as simple as using brackets instead of dot syntax.

Reply:
It's a 3-prong approach...

- `mvkOSVersionIsAtLeast()` or `supportsFeatureSet:` are used when something such as a `MTLPixelFormat`, other enum value, numeric limit,  etc are currently available during compilation, but not available in older runtime OS versions.

- `respondsToSelector:` is used in a similar situation specifically when a method is available on an OS currently and so compiles, but wasn't available in earlier OS versions, and so needs to be guarded against older runtimes. It's a more definitive test than the previous option.

- `#if MVK_TVOS`, etc, are used where a method, `MTLPixelFormat`, enum value, etc. is not available at all on an OS, and won't even pass compilation.

And of course, occasionally we need to combine these techniques.

Reply:
Thanks for the clarity here, it is very helpful!